renku/ui/cli/dataset.py

```diff
@@ -934,12 +949,44 @@ def export_(name, provider, publish, tag, **kwargs):
     click.secho("OK", fg=color.GREEN)
 
 
+def import_provider_options(*param_decls, **attrs):
+    """Sets dataset import provider option groups on the dataset import command."""
+
+    def wrapper(f):
+        from click_option_group import optgroup
+
+        from renku.core.dataset.providers import ProviderFactory
+
+        providers = [
+            (k, v) for k, v in ProviderFactory.providers().items() if v.supports_import and v.import_parameters()
+        ]
+
+        for i, (name, provider) in enumerate(providers):
+            params = provider.import_parameters()
+
+            for j, (param_name, (param_description, param_type)) in enumerate(params.items()):
+                if j == 0:
+                    param_description = f"\b\n{param_description}\n "  # NOTE: add newline after a group
+                f = optgroup.option(f"--{param_name}", type=param_type, help=param_description, default=None)(f)
+
+            name = f"{name} configuration"
+            if i == len(providers) - 1:
+                name = "\n  " + name  # NOTE: add newline before first group
+
+            f = optgroup.group(name=name)(f)
```

Comment:
Probably something for a separate issue, but it'd be really nice if we could return a `Proxy` here, for import times. If each proxy returns a wrapped Proxy unless there's no `next()` that should work I think, and only be evaluated if this method is actually accessed. I.e. Proxy wrapping the whole decorator instead of just wrapping a value inside the decorator. But I'm not sure if click accepts that